[kind/bug-report] Issue Title

**Describe the bug**
SpiderFoot is not installed (or not functional) out of the box on the current Trace Labs OSINT VM. Manual installation fails because the VM ships with **Python 3.13**, which is incompatible with the `lxml>=4.9.2,<5` pin in SpiderFoot's `requirements.txt`. The net effect is that this tool is not usable on the VM

**To Reproduce**
Steps to reproduce:
1. Boot the current Trace Labs OSINT VM.
2. Try to launch SpiderFoot as on prior VM versions (it isn't installed/working).
3. Install it manually — activate the venv and `pip install -r /opt/spiderfoot/requirements.txt`.
4. Watch the dependency install fail on lxml.

**Expected behavior**
SpiderFoot is present and runnable on the VM out of the box or a clean manual `pip install -r requirements.txt` completes successfully and launches.

**Actual behavior**
SpiderFoot is missing dependencies, and the manual install can't complete because lxml won't install:
- The `lxml>=4.9.2,<5` pin forces an lxml 4.9.x version. No lxml in the 4.9.x line builds against Python 3.13, so pip tries to compile from source and fails with CPython-3.13 API errors:
  - `implicit declaration of function '_PyInterpreterState_GetConfig'`
  - `too few arguments to function '_PyLong_AsByteArray'`
  - `'_PyObject_NextNotImplemented' undeclared`
  - `implicit declaration of '_PyGen_SetStopIterationValue' / '_PySet_NextEntry' / '_PyUnicode_FastCopyCharacters'`
  - ending in `error: command '/usr/bin/x86_64-linux-gnu-gcc' failed with exit code 1` → `Failed to build lxml`.
- Secondary friction encountered while trying to recover the manual install:
  - `/opt/spiderfoot` and its venv are **root-owned**, so installing as the `osint` user fails mid-way: `OSError: [Errno 13] Permission denied: '/opt/spiderfoot/venv/lib/python3.13/site-packages/gexf'`.
  - Falling back to `sudo pip` doesn't use the venv (`sudo` resets PATH), so Debian 13 blocks it with `error: externally-managed-environment` (PEP 668).

**Environment Info**
- Host OS: Windows 11
- VM software & version: 
- Guest: TLOSINT-VM 2026.05 full

**Screenshots / Logs**
```
src/lxml/etree.c:5988: error: implicit declaration of function '_PyInterpreterState_GetConfig'
src/lxml/etree.c:269066: error: too few arguments to function '_PyLong_AsByteArray'
src/lxml/etree.c:265182: error: '_PyObject_NextNotImplemented' undeclared
...
error: command '/usr/bin/x86_64-linux-gnu-gcc' failed with exit code 1
ERROR: Failed building wheel for lxml
Failed to build lxml

# secondary failures while recovering the manual install
OSError: [Errno 13] Permission denied: '/opt/spiderfoot/venv/lib/python3.13/site-packages/gexf'
error: externally-managed-environment
```

**Additional context**
- **Root cause of the install failure:** the `lxml<5` upper bound is incompatible with Python 3.13. Bumping it to `lxml>=5.2` (or dropping `,<5`) lets pip pull a prebuilt manylinux wheel for 3.13 and skips compilation entirely.

  - Debian 13 nudges toward **pipx** for Python apps, but pipx alone wouldn't fix this — the `lxml<5` pin would still fail to build on 3.13. The dependency pin must be corrected regardless.
  
## Temporary Workaround:

Run this 1-liner to get Spiderfoot up and running if needed before I have a chance to fix this: 
  `sudo apt install -y libxml2-dev libxslt1-dev && sudo sed -i -E 's/^(\s*lxml>=[0-9.]+),<5/\1/' /opt/spiderfoot/requirements.txt && sudo chown -R osint:osint /opt/spiderfoot && source /opt/spiderfoot/venv/bin/activate && pip install -r /opt/spiderfoot/requirements.txt && deactivate`

--- a/scripts/tlosint-tools.sh
+++ b/scripts/tlosint-tools.sh
@@ -2,6 +2,7 @@
 # shellcheck disable=SC1071
 # Ultimate OSINT Setup for Debian + Updater + Validator
 # 2025-09-09: fix SpiderFoot venv installer (zsh + set -u), keep Firefox hardening, PATH fix, Shodan deferred OK, StegOSuite optional.
+# 2026-07-08: relax SpiderFoot's lxml<5 pin so pip uses a prebuilt lxml 5.x wheel on Python 3.13 (Debian 13) instead of failing to compile.
 set -uo pipefail
 export DEBIAN_FRONTEND=noninteractive
 
@@ -184,8 +185,21 @@ install_spiderfoot_from_source() {
   local venv="${dest}/venv"
 
   log "[*] Installing SpiderFoot from source into ${dest}"
+  # lxml build deps: only used as a fallback if pip has to compile from source.
+  apt_try_install libxml2-dev || true
+  apt_try_install libxslt1-dev || true
   run "${SUDO} rm -rf \"${dest}\""
   run "${SUDO} git clone --depth=1 \"${repo}\" \"${dest}\""
+  # Drop the upstream 'lxml>=4.9.2,<5' upper bound. lxml 4.9.x has no prebuilt
+  # wheel for Python 3.13 and fails to compile against the 3.13 C-API, so pip
+  # aborts. Removing the '<5' cap lets pip pull a prebuilt manylinux wheel for
+  # 3.13. The regex tolerates whitespace and a '<5.0'-style cap; it only edits
+  # the copy under /opt (upstream repo untouched).
+  if [[ -f "${dest}/requirements.txt" ]] \
+     && ${SUDO} grep -Eq '^[[:space:]]*lxml[[:space:]]*>=[[:space:]]*[0-9.]+[[:space:]]*,[[:space:]]*<[[:space:]]*5' "${dest}/requirements.txt"; then
+    log "[*] Relaxing SpiderFoot lxml '<5' pin for Python 3.13 compatibility"
+    run "${SUDO} sed -i -E 's/^([[:space:]]*lxml[[:space:]]*>=[[:space:]]*[0-9.]+)[[:space:]]*,[[:space:]]*<[[:space:]]*5[0-9.]*/\\1/' \"${dest}/requirements.txt\""
+  fi
   run "${SUDO} python3 -m venv \"${venv}\""
   run "${SUDO} \"${venv}/bin/pip\" -q install --upgrade pip wheel setuptools"
   if [[ -f "${dest}/requirements.txt" ]]; then

